spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/MetaStore.scala

```diff
@@ -0,0 +1,573 @@
+      |From ${getMetaStoreTableFullName}
+    """.stripMargin.replaceAll("\n", " ")
+    val names = ListBuffer[String]()
+    // Add source tables from metastore
+    metaStoreConn.withSessionDo {
+      session =>
+        val result = session.execute(selectQuery).iterator()
+        while (result.hasNext) {
+          val row: Row = result.next()
+          val clusterName = row.getString(0)
+          names += clusterName
+        }
+    }
+
+    names.toList
+  }
+
+  /** Store a tale with the creation meta data */
+  override def storeTable(
+      tableIdent: TableIdent,
+      source: String,
+      schema: Option[StructType],
+      options: Map[String, String]): Unit = {
+    import collection.JavaConversions._
+
+    val cluster = tableIdent.cluster.getOrElse(sqlContext.getCluster)
+    if (schema.nonEmpty) {
+      metaStoreConn.withSessionDo {
+        session =>
+          val preparedStatement = session.prepare(InsertIntoMetaStoreQuery)
+          session.execute(
+            preparedStatement.bind(
+              cluster,
+              tableIdent.keyspace,
+              tableIdent.table,
+              source,
+              schema.get.json,
+              mapAsJavaMap(options)))
+      }
+    } else {
+      metaStoreConn.withSessionDo {
+        session =>
+          val preparedStatement: PreparedStatement = session.prepare(InsertIntoMetaStoreWithoutSchemaQuery)
+          session.execute(
+            preparedStatement.bind(
+              cluster,
+              tableIdent.keyspace,
+              tableIdent.table,
+              source,
+              mapAsJavaMap(options)))
+      }
+    }
+
+    // Remove temporary database or table
+    val tempTableIdent = TableIdent(
+      TempDatabaseOrTableToBeDeletedName,
+      TempDatabaseOrTableToBeDeletedName,
+      Option(cluster))
+    removeTable(tempTableIdent)
+  }
+
+  override def setTableOption(tableIdent: TableIdent, key: String, value: String) : Unit = {
+    val updateQuery =
+      s"""
+      |UPDATE ${getMetaStoreTableFullName}
+      |SET options['$key'] = '$value'
+      |WHERE cluster_name = '${tableIdent.cluster.get}'
+      | AND keyspace_name = '${tableIdent.keyspace}'
+      | AND table_name = '${tableIdent.table}'
+    """.stripMargin.replaceAll("\n", " ")
+    metaStoreConn.withSessionDo {
+      session => session.execute(updateQuery)
+    }
+  }
+
+  override def removeTableOption(tableIdent: TableIdent, key: String) : Unit = {
+    val deleteQuery =
+      s"""
+      |DELETE options['$key']
+      |FROM ${getMetaStoreTableFullName}
+      |WHERE cluster_name = '${tableIdent.cluster.get}'
+      | AND keyspace_name = '${tableIdent.keyspace}'
+      | AND table_name = '${tableIdent.table}'
+    """.stripMargin.replaceAll("\n", " ")
+    metaStoreConn.withSessionDo {
+      session => session.execute(deleteQuery)
+    }
+  }
+
+  /** Set table schema */
+  override def setTableSchema(tableIdent: TableIdent, scheamJsonString: String) : Unit = {
```

Comment:
typo (the same typo found also in other places - I'm not listing all of them in the comments)